ARCL/Source/LocationNode.swift

```diff
@@ -89,8 +89,32 @@ open class LocationAnnotationNode: LocationNode {
 
         addChildNode(annotationNode)
     }
+    
+    // Use this constructor to add a UIView as an annotation
+    // UIView is more configurable then a UIImage allowing to add background image and labels
+    public convenience init(location: CLLocation?, markerView: UIView){
+        self.init(location: location, image: markerView.image() ?? UIImage())
+    }
 
     required public init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 }
+
+extension UIView{
```

Comment:
As this library supports iOS 11 and above we can conveniently use the newly introduced apis to convert View hierarchy to Image

you can re-write this extension (as I have used in my  version of code)

```
public func renderToImage(afterScreenUpdates: Bool = true) -> UIImage {
        let rendererFormat = UIGraphicsImageRendererFormat.default()
        rendererFormat.opaque = false
        let renderer = UIGraphicsImageRenderer(size: bounds.size, format: rendererFormat)
        
        let snapshotImage = renderer.image { _ in
            drawHierarchy(in: bounds, afterScreenUpdates: afterScreenUpdates)
        }
        return snapshotImage
    }
```

Reply:
I reviewed your code. I see what you are trying to do. Will update my PR with the necessary changes and tests.